cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -126,6 +128,39 @@ public void deleteRepository(FullHttpRequest request, HttpResponder responder,
                                                               namespaceId));
   }
 
+  /**
+   * Lists applications found in linked repository for the given namespace.
+   * Returns the application name and git-filehash for each application config file found
+   * <pre>
+   * {@code
+   * {
+   *  apps:[
```

Comment:
since this is just a simple list, say returning 5000 pipelines should be fine right?

Reply:
Yeah. It would be ~1.5 MB but 5000 pipelines will be rare. We can consider pagination later if we see more usage with high no of pipelines

Reply:
I am also concerned of the time this call would take, not only data amount. If we go with non-paginated approach, pleas create a JIRA to perform a scale test of this API